Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.h

```diff
@@ -0,0 +1,177 @@
+    Data<bool> d_computeHexahedra; ///< If true, will compute hexahedra list and index list inside the ROI. (default = true)
+    Data<bool> d_strict; ///< If true, an element is inside the box if all of its nodes are inside. If False, only the center point of the element is checked. (default = true)
+
+    //Output
+    Data<SetIndex> d_indices; ///< Indices of the points contained in the ROI
+    Data<SetIndex> d_edgeIndices; ///< Indices of the edges contained in the ROI
+    Data<SetIndex> d_triangleIndices; ///< Indices of the triangles contained in the ROI
+    Data<SetIndex> d_quadIndices; ///< Indices of the quad contained in the ROI
+    Data<SetIndex> d_tetrahedronIndices; ///< Indices of the tetrahedra contained in the ROI
+    Data<SetIndex> d_hexahedronIndices; ///< Indices of the hexahedra contained in the ROI
+    Data<VecCoord> d_pointsInROI; ///< Points contained in the ROI
+    Data<VecEdge> d_edgesInROI; ///< Edges contained in the ROI
+    Data<VecTriangle> d_trianglesInROI; ///< Triangles contained in the ROI
+    Data<VecQuad> d_quadsInROI; ///< Quad contained in the ROI
+    Data<VecTetra> d_tetrahedraInROI; ///< Tetrahedra contained in the ROI
+    Data<VecHexa> d_hexahedraInROI; ///< Hexahedra contained in the ROI
+    Data< sofa::Size > d_nbIndices; ///< Number of selected indices
+
+    Data<VecCoord > d_pointsOutROI; ///< Points not contained in the ROI
+    Data<type::vector<Edge> > d_edgesOutROI; ///< Edges not contained in the ROI
+    Data<type::vector<Triangle> > d_trianglesOutROI; ///< Triangles not contained in the ROI
+    Data<type::vector<Quad> > d_quadsOutROI; ///< Quads not contained in the ROI
+    Data<type::vector<Tetra> > d_tetrahedraOutROI; ///< Tetrahedra not contained in the ROI
+    Data<type::vector<Hexa> > d_hexahedraOutROI; ///< Hexahedra not contained in the ROI
+    Data<SetIndex> d_indicesOut; ///< Indices of the points not contained in the ROI
+    Data<SetIndex> d_edgeOutIndices; ///< Indices of the edges not contained in the ROI
+    Data<SetIndex> d_triangleOutIndices; ///< Indices of the triangles not contained in the ROI
+    Data<SetIndex> d_quadOutIndices; ///< Indices of the quads not contained in the ROI
+    Data<SetIndex> d_tetrahedronOutIndices; ///< Indices of the tetrahedra not contained in the ROI
+    Data<SetIndex> d_hexahedronOutIndices; ///< Indices of the hexahedra not contained in the ROI
+
+    //Parameter
+    Data<bool> d_drawROI; ///< Draw the ROI. (default = false)
+    Data<bool> d_drawPoints; ///< Draw Points. (default = false)
+    Data<bool> d_drawEdges; ///< Draw Edges. (default = false)
+    Data<bool> d_drawTriangles; ///< Draw Triangles. (default = false)
+    Data<bool> d_drawQuads; ///< Draw Quads. (default = false)
+    Data<bool> d_drawTetrahedra; ///< Draw Tetrahedra. (default = false)
+    Data<bool> d_drawHexahedra; ///< Draw Tetrahedra. (default = false)
+    Data<double> d_drawSize; ///< rendering size for ROI and topological elements
```

Comment:
```suggestion
    Data<float> d_drawSize; ///< rendering size for ROI and topological elements
```
Since it is used in the drawing API